buildconfig/CMake/FindSIP.cmake

```diff
@@ -1,78 +1,107 @@
+  The path to the sip-build executable
+
+``SIP_MODULE_EXECUTABLE``
+  The path to the sip-module executable
+
+For versions prior to 5 this file additionally defines the following variables:
+
+``SIP_EXECUTABLE``
+Path and filename of the SIP command line executable.
+
+``SIP_INCLUDE_DIR``
+Directory holding the SIP C++ header file.
+
+#]=======================================================================]
+include(FindPackageHandleStandardArgs)
+
+# First look for sip-build, indicating the newer v6 build system
+find_program(SIP_BUILD_EXECUTABLE sip-build)
+if(SIP_BUILD_EXECUTABLE)
+  # version string
+  execute_process(
+    COMMAND ${SIP_BUILD_EXECUTABLE} --version
+    OUTPUT_VARIABLE SIP_VERSION
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
+
+  # module generator
+  find_program(SIP_MODULE_EXECUTABLE sip-module)
+
+  # pyproject.toml template
+  find_file(SIP_PYPROJECT_TOML_TEMPLATE NAME pyproject.toml.in
+            PATHS ${CMAKE_MODULE_PATH}/sip-build
+  )
+
+  # project.py template
+  find_file(SIP_PROJECT_PY_TEMPLATE NAME project.py.in
+            PATHS ${CMAKE_MODULE_PATH}/sip-build
+  )
+
+  # Set expected variables for find_package
+  find_package_handle_standard_args(
+    SIP
+    REQUIRED_VARS SIP_BUILD_EXECUTABLE SIP_MODULE_EXECUTABLE
+                  SIP_PYPROJECT_TOML_TEMPLATE SIP_PROJECT_PY_TEMPLATE
+    VERSION_VAR SIP_VERSION
+  )
+endif()
 
-  EXECUTE_PROCESS(COMMAND ${Python_EXECUTABLE} ${_find_sip_py} OUTPUT_VARIABLE sip_config)
-  IF(sip_config)
-    STRING(REGEX REPLACE "^sip_version:([^\n]+).*$" "\\1" SIP_VERSION ${sip_config})
-    STRING(REGEX REPLACE ".*\nsip_version_str:([^\n]+).*$" "\\1" SIP_VERSION_STR ${sip_config})
-    STRING(REGEX REPLACE ".*\nsip_bin:([^\n]+).*$" "\\1" SIP_EXECUTABLE ${sip_config})
-    IF(NOT SIP_DEFAULT_SIP_DIR)
-        STRING(REGEX REPLACE ".*\ndefault_sip_dir:([^\n]+).*$" "\\1" SIP_DEFAULT_SIP_DIR ${sip_config})
-    ENDIF(NOT SIP_DEFAULT_SIP_DIR)
-    STRING(REGEX REPLACE ".*\nsip_inc_dir:([^\n]+).*$" "\\1" SIP_INCLUDE_DIRECTORY ${sip_config})
-    SET(SIP_FOUND TRUE)
-  ENDIF(sip_config)
-
-  # Check that it really exists
-  if(${CMAKE_SYSTEM} MATCHES "Darwin" AND IS_SYMLINK ${SIP_INCLUDE_DIRECTORY}/sip.h)
-    get_filename_component(_sip_h "${SIP_INCLUDE_DIRECTORY}/sip.h" REALPATH BASE_DIR)
-    get_filename_component(SIP_INCLUDE_DIRECTORY "${_sip_h}" DIRECTORY)
+if(NOT SIP_FOUND)
+  # Python development is required for the older system
+  if(NOT Python_Development_FOUND)
+    message(
+      FATAL_ERROR "FindSIP requires find_package(Python) to be called first"
+    )
   endif()
-  find_path(SIP_INCLUDE_DIR sip.h PATHS ${SIP_INCLUDE_DIRECTORY} NO_DEFAULT_PATH)
 
-  include ( FindPackageHandleStandardArgs )
-  find_package_handle_standard_args( SIP DEFAULT_MSG SIP_VERSION_STR SIP_INCLUDE_DIR SIP_EXECUTABLE )
+  # Look for older sip build system. CentOS has this prefixed with python3
+  find_program(SIP_EXECUTABLE NAMES python${Python_VERSION_MAJOR}-sip sip)
```

Comment:
Can this be set to `REQUIRED` since were already on the fallback, or is find_package in the parent marked as required?

Reply:
Yep the PyQt/sip finders set `REQUIRED` when they care called: https://github.com/mantidproject/mantid/blob/9e22be693950e5cf1c768578f7d725fd9d2d29fa/CMakeLists.txt#L120